### ISSUE TITLE
Pulse 14: will Manjaro/Arch get supported?

### ISSUE THREAD

**Issue by Shmuji7** [tuxedocomputers/tuxedo-control-center#90]
Hi there,

I'm owning this wonderful little since a while now but am not using it because of the fan sound starting with the slightest load. BIOS fan setting seem not to work and support is suggesting switching to Ubuntu. For me using Manjaro since some years now this is not an option :-)

Helmut Stult is maintaining an installer package within the Manjaro 'extra' repo (thank you Helmut!!). However, reading CPU temperature and controlling fan speed are not supported for my device (options ar blurred).

Are you planning to add support for the Pulse 14 under Manjaro? Would you know a manual work around for me?

Thanks a lot!

**Comment by tuxedoxt** [comment]
Hello,

you will need the `tuxedo-keyboard` driver package as well. We don't package for Majaro & Arch directly. As you discovered there are still packages to be found. I believe if you found a tuxedo-control-center package, a tuxedo-keyboard one should not be far away ;)

**Comment by Shmuji7** [comment]
**TL;DR:** it's working now, thank you a lot! Such a relief!
`tuxedo-keyboard` is requiring package `linux-headers`. The latter has not gotten installed automatically by the GUI software installer as there are alternative versions.


Extended version:
That's been quick, thanks a lot!

I wanted to be as quick myself, starting to write that  `tuxedo-keyboard` is included as dependency in Helmut's package already, To prove it I've been reinstalling the package and copied the log:
```
Preparing...
Synchronizing package databases...
Refreshing core.db...
Resolving dependencies...
Checking inter-conflicts...
Download of grub (2.04-21) started
Download of grub (2.04-21) finished
Checking keyring...
Checking integrity...
Loading packages files...
Checking file conflicts...
Checking available disk space...
Upgrading grub (2.04-19 -> 2.04-21)...
Installing tuxedo-keyboard (3.0.2-1)...
Installing tuxedo-control-center (1.0.11-1)...
Created symlink /etc/systemd/system/multi-user.target.wants/tccd.service → /etc/systemd/system/tccd.service.
Created symlink /etc/systemd/system/sleep.target.wants/tccd-sleep.service → /etc/systemd/system/tccd-sleep.service.
Running post-transaction hooks...
Arming ConditionNeedsUpdate...
Install DKMS modules
==> Unable to install module tuxedo-keyboard/3.0.2 for kernel 5.10.19-1-MANJARO: Missing kernel headers.
==> Unable to install module tuxedo-keyboard/3.0.2 for kernel 5.9.16-1-MANJARO: Missing kernel headers.
Reloading system bus configuration...
Updating icon theme caches...
Updating the info directory file...
Updating the desktop file MIME type cache...
Transaction successfully finished.
```
Though using professional Manjaro I'm no hardcore geek. `Transaction successfully finished.` made me think that everyhing's OK. But before posting the log to you, I've glanced the middle piece and hey, what's this: `Unable to install module tuxedo-keyboard/3.0.2 for kernel 5.10.19-1-MANJARO: Missing kernel headers.` ???
What are kernel headers? Something to eat? :-) And why hasn't the Manjaro graphical "Add/Remove Software" tool I'm using reported an error during installation?

Crosschecking with "Add/Remove Software" and paying more attention do details I've realized that even this tool is offering kernel headers for 10 different kernel versions from 4.4 to 5.11 as optional dependencies. Maybe this is a rare example of a functional gap for the graphical Manjaro installer when there are multiple options for one mandatory dependency. I'll let Helmut know, curious what he'll say.

Anyway: with help from [Korbin Brown's posting](https://linuxconfig.org/manjaro-linux-kernel-headers-installation) explaining how to deal with kernel headers one command `pacman -S linux-headers` saved the day, week, month for me:
```
sudo pacman -S linux-headers
[sudo] password for p: 
:: There are 10 providers available for linux-headers:
:: Repository core
   1) linux414-headers  2) linux419-headers  3) linux44-headers
   4) linux49-headers  5) linux510-headers  6) linux511-headers
   7) linux54-headers  8) linux59-headers
:: Repository community
   9) linux54-rt-headers  10) linux59-rt-headers

Enter a number (default=1): 5
resolving dependencies...
looking for conflicting packages...

Packages (1) linux510-headers-5.10.19-1

Total Download Size:    9,46 MiB
Total Installed Size:  47,72 MiB

:: Proceed with installation? [Y/n] 
:: Retrieving packages...
 linux510-headers...     9,5 MiB  93,4 KiB/s 01:44 [##################] 100%
(1/1) checking keys in keyring                     [##################] 100%
(1/1) checking package integrity                   [##################] 100%
(1/1) loading package files                        [##################] 100%
(1/1) checking for file conflicts                  [##################] 100%
(1/1) checking available disk space                [##################] 100%
:: Processing package changes...
(1/1) installing linux510-headers                  [##################] 100%
:: Running post-transaction hooks...
(1/3) Arming ConditionNeedsUpdate...
(2/3) Updating module dependencies...
(3/3) Install DKMS modules
==> dkms install --no-depmod -m tuxedo-keyboard -v 3.0.2 -k 5.10.19-1-MANJARO
==> depmod 5.10.19-1-MANJARO
```
So happy! THANK YOU SO MUCH FOR BEING SO RESPONSIVE, encouraging me to dig deeper myself!
